Kmodify test failure due to signed/unsigned char

On my Oracle Linux 10 laptop I'm getting the following failures on  6.17.7-vmtest38.1default for x86_64:

```
======================================================================
ERROR: test_many_out_params (tests.linux_kernel.helpers.test_kmodify.TestCallFunction.test_many_out_params)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/stepbren/repos/drgn/tests/linux_kernel/helpers/test_kmodify.py", line 242, in test_many_out_params
    self.assert_called("many_out_params", args)
  File "/home/stepbren/repos/drgn/tests/linux_kernel/helpers/test_kmodify.py", line 34, in assert_called
    return_value = call_function(self.prog["drgn_kmodify_test_" + name], *args)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/stepbren/repos/drgn/drgn/helpers/common/prog.py", line 115, in wrapper
    return f(args[0].prog_, *args, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/stepbren/repos/drgn/drgn/helpers/experimental/kmodify.py", line 1501, in call_function
    implicit_convert(
TypeError: cannot convert 'char *' to incompatible type 'char *'

======================================================================
ERROR: test_string_args (tests.linux_kernel.helpers.test_kmodify.TestCallFunction.test_string_args) [array]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/stepbren/repos/drgn/tests/linux_kernel/helpers/test_kmodify.py", line 183, in test_string_args
    self.assert_called(
  File "/home/stepbren/repos/drgn/tests/linux_kernel/helpers/test_kmodify.py", line 34, in assert_called
    return_value = call_function(self.prog["drgn_kmodify_test_" + name], *args)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/stepbren/repos/drgn/drgn/helpers/common/prog.py", line 115, in wrapper
    return f(args[0].prog_, *args, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/stepbren/repos/drgn/drgn/helpers/experimental/kmodify.py", line 1501, in call_function
    implicit_convert(
TypeError: cannot convert 'char *' to incompatible type 'char *'

======================================================================
ERROR: test_string_args (tests.linux_kernel.helpers.test_kmodify.TestCallFunction.test_string_args) [array pointer]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/stepbren/repos/drgn/tests/linux_kernel/helpers/test_kmodify.py", line 183, in test_string_args
    self.assert_called(
  File "/home/stepbren/repos/drgn/tests/linux_kernel/helpers/test_kmodify.py", line 34, in assert_called
    return_value = call_function(self.prog["drgn_kmodify_test_" + name], *args)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/stepbren/repos/drgn/drgn/helpers/common/prog.py", line 115, in wrapper
    return f(args[0].prog_, *args, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/stepbren/repos/drgn/drgn/helpers/experimental/kmodify.py", line 1501, in call_function
    implicit_convert(
TypeError: cannot convert 'char *' to incompatible type 'char *'

----------------------------------------------------------------------
```

This is not reproducing on my Arch system or on the Debian rootfs. When running the tests with `--pdb` I get to the root of the issue:

```
            if isinstance(arg, pass_pointer):
                if not isinstance(arg.object, Object):
                    arg.object = Object(prog, value=arg.object)
                type = arg.object.type_
                unaliased_type = type.unaliased()
                if unaliased_type.kind == TypeKind.ARRAY:
                    type = unaliased_type.type
                if i < len(func_type.parameters):
                    # We don't need the result, just type checking.
>                   implicit_convert(
                        func_type.parameters[i].type,
                        Object(prog, prog.pointer_type(type), 0),
                    )
E                   TypeError: cannot convert 'char *' to incompatible type 'char *'

drgn/helpers/experimental/kmodify.py:1501: TypeError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/stepbren/repos/drgn/drgn/helpers/experimental/kmodify.py(1501)call_function()
-> implicit_convert(
(Pdb) p func_type.parameters[i].type
prog.pointer_type(type=prog.int_type(name='char', size=1, is_signed=False))
(Pdb) p prog.pointer_type(type)
prog.pointer_type(type=prog.int_type(name='char', size=1, is_signed=True))
(Pdb) p prog.type("char")
prog.int_type(name='char', size=1, is_signed=True)
```

Compare this to my Arch system:
```
>>> prog["UTS_RELEASE"]
(const char [25])"6.17.7-vmtest38.1default"
>>> print(repr(prog.type("char")))
prog.int_type(name='char', size=1, is_signed=False)
```

--- a/libdrgn/debug_info.c
+++ b/libdrgn/debug_info.c
@@ -1580,6 +1580,14 @@ drgn_module_maybe_use_elf_file(struct drgn_module *module,
 		prog->tried_main_language = false;
 		module->elf_symtab_pending_files |=
 			DRGN_MODULE_FILE_MASK_DEBUG;
+		// In case primitive types have been created by drgn before
+		// loading the main debug file (e.g. by reading VMCOREINFO),
+		// clear this cache so we can look them up from the debug info.
+		// See https://github.com/osandov/drgn/issues/569 for more
+		// details.
+		if (module->kind == DRGN_MODULE_MAIN)
+			memset(module->prog->primitive_types, 0,
+			       sizeof(module->prog->primitive_types));
 	}
 	if (!prog->has_platform) {
 		drgn_log_debug(prog, "setting program platform from %s",

